client.go

```diff
@@ -182,6 +182,18 @@ func (c *Client[Req, Res]) CallClientStream(ctx context.Context) *ClientStreamFo
 	}
 }
 
+// CallClientStream calls a client streaming procedure in simple mode.
+func (c *Client[Req, Res]) CallClientStreamSimple(ctx context.Context) (*ClientStreamForClient[Req, Res], error) {
```

Comment:
I don't think we want to change the signature here to return an error, but I'll leave that up to others. I would expect this to just return a stream (same with bidi). Guessing that you added this so you could do the implicit `stream.Send(nil)`.

I think we discussed even removing the `stream.RequestHeader` function if you're using the simple API, but not sure if that's still a thing.

Reply:
I did this intentionally. Personally, I think it's the right thing to do. The main reason why we made the client stream and bidi stream calls infallible is because they defer the connection until after the first message is sent so that it is possible to set headers/etc. However, with the simple API, this is no longer needed, so client stream and bidi stream can now be symmetrical with unary and server stream.